ext/nnstreamer/android_source/gstamcsrc.c

```diff
@@ -0,0 +1,986 @@
+    }
+
+    priv->filename = g_strdup (location);
+    return gst_amc_src_media_open (self);
+  }
+
+  LOGE ("A file location should be provided");
+  return FALSE;
+}
+
+/**
+ * @brief set amcsrc properties
+ */
+static void
+gst_amc_src_set_property (GObject * object, guint prop_id,
+    const GValue * value, GParamSpec * pspec)
+{
+  GstAMCSrc *self = GST_AMC_SRC (object);
+
+  switch (prop_id) {
+    case PROP_LOCATION:
+      gst_amc_src_set_location (self, g_value_get_string (value), NULL);
+      break;
+    default:
+      G_OBJECT_WARN_INVALID_PROPERTY_ID (object, prop_id, pspec);
+      break;
+  }
+}
+
+/**
+ * @brief get amcsrc properties
+ */
+static void
+gst_amc_src_get_property (GObject * object, guint prop_id, 
+    GValue *value, GParamSpec * pspec)
+{
+  GstAMCSrc *self = GST_AMC_SRC (object);
+  GstAMCSrcPrivate *priv = GST_AMC_SRC_GET_PRIVATE (self);
+
+  switch (prop_id) {
+    case PROP_LOCATION:
+      g_value_set_string (value, priv->filename);
+      break;
+    default:
+      G_OBJECT_WARN_INVALID_PROPERTY_ID (object, prop_id, pspec);
+      break;
+  }
+}
+
+/**
+ * @brief unlock function, flush any pending data in the data queue
+ */
+static gboolean
+gst_amc_src_unlock (GstBaseSrc * src)
+{
+  GstAMCSrc *self = GST_AMC_SRC (src);
+  GstAMCSrcPrivate *priv = GST_AMC_SRC_GET_PRIVATE (self);
+
+  LOGI ("unlocking create");
+  gst_data_queue_set_flushing (priv->outbound_queue, TRUE);
+
+  return TRUE;
+}
+
+/**
+ * @brief unlock_stop function, clear the previous unlock request
+ */
+static gboolean
+gst_amc_src_unlock_stop (GstBaseSrc * src)
+{
+  GstAMCSrc *self = GST_AMC_SRC (src);
+  GstAMCSrcPrivate *priv = GST_AMC_SRC_GET_PRIVATE (self);
+
+  LOGI ("stopping unlock");
+  gst_data_queue_set_flushing (priv->outbound_queue, FALSE);
+
+  return TRUE;
+}
+
+/**
+ * @brief get caps of subclass
+ */
+static GstCaps *
+gst_amc_src_get_caps (GstBaseSrc * src, GstCaps * filter)
+{
+  GstAMCSrc *self = GST_AMC_SRC (src);
+  GstAMCSrcPrivate *priv = GST_AMC_SRC_GET_PRIVATE (self);
+  GstStructure *format;
+  GstPad *pad = src->srcpad;
+  GstCaps *current_caps = gst_pad_get_current_caps (pad);
+  GstCaps *caps = gst_caps_new_empty ();
+
+  GST_OBJECT_LOCK (self);
+
+  /** width, height, and framerate were obtained from the media format */
+  format = gst_structure_new ("video/x-raw",
+      "format", G_TYPE_STRING, "NV12",  /** TODO Support other formats? */
+      "width", G_TYPE_INT, priv->width,
+      "height", G_TYPE_INT, priv->height,
+      "interlaced", G_TYPE_BOOLEAN, FALSE,
+      "pixel-aspect-ratio", GST_TYPE_FRACTION, 1, 1,
+      "framerate", GST_TYPE_FRACTION, priv->framerate, 1, NULL);
+
+  gst_caps_append_structure (caps, format);
+
+  if (current_caps) {
+    GstCaps *intersection =
+      gst_caps_intersect_full (current_caps, caps, GST_CAPS_INTERSECT_FIRST);
+
+    gst_caps_unref (current_caps);
+    gst_caps_unref (caps);
+    caps = intersection;
+  }
+
+  gst_structure_free (format);
```

Comment:
plz check this line.
gst_caps_append_structure() takes the structure and it does not need to free.

Reply:
Right... 'gst_caps_append_structure' does not copy 'format'. I will fix this.